includes/class-compatibility-service.php

```diff
@@ -76,10 +76,10 @@ public function add_compatibility_onboarding_data( $args ): array {
 	private function get_compatibility_data(): array {
 		$active_plugins        = get_option( 'active_plugins', [] );
 		$post_types_count      = $this->get_post_types_count();
-		$wc_permalinks         = get_option( 'woocommerce_permalinks' );
-		$wc_shop_permalink     = get_permalink( wc_get_page_id( 'shop' ) );
-		$wc_cart_permalink     = get_permalink( wc_get_page_id( 'cart' ) );
-		$wc_checkout_permalink = get_permalink( wc_get_page_id( 'checkout' ) );
+		$wc_permalinks         = get_option( 'woocommerce_permalinks', [] );
+		$wc_shop_permalink     = get_permalink( wc_get_page_id( 'shop' ) ) ? get_permalink( wc_get_page_id( 'shop' ) ) : 'Not set';
```

Comment:
What do you think about extracting that repeating logic to a method? My idea is to have something like this:

<details><summary>Details</summary>

```diff
diff --git a/includes/class-compatibility-service.php b/includes/class-compatibility-service.php
index bd831fd52..6a66941a1 100644
--- a/includes/class-compatibility-service.php
+++ b/includes/class-compatibility-service.php
@@ -77,9 +77,9 @@ class Compatibility_Service {
 		$active_plugins        = get_option( 'active_plugins', [] );
 		$post_types_count      = $this->get_post_types_count();
 		$wc_permalinks         = get_option( 'woocommerce_permalinks', [] );
-		$wc_shop_permalink     = get_permalink( wc_get_page_id( 'shop' ) ) ? get_permalink( wc_get_page_id( 'shop' ) ) : 'Not set';
-		$wc_cart_permalink     = get_permalink( wc_get_page_id( 'cart' ) ) ? get_permalink( wc_get_page_id( 'cart' ) ) : 'Not set';
-		$wc_checkout_permalink = get_permalink( wc_get_page_id( 'checkout' ) ) ? get_permalink( wc_get_page_id( 'checkout' ) ) : 'Not set';
+		$wc_shop_permalink     = $this->get_permalink_for_page_id( 'shop' );
+		$wc_cart_permalink     = $this->get_permalink_for_page_id( 'cart' );
+		$wc_checkout_permalink = $this->get_permalink_for_page_id( 'checkout' );
 
 		return [
 			'woopayments_version'    => WCPAY_VERSION_NUMBER,
@@ -114,4 +114,17 @@ class Compatibility_Service {
 
 		return $post_types_count;
 	}
+
+	/**
+	 * Gets the permalink for a page ID.
+	 *
+	 * @param string $page_id The page ID to get the permalink for.
+	 *
+	 * @return string The permalink for the page ID, or 'Not set' if the permalink is not available.
+	 */
+	private function get_permalink_for_page_id( string $page_id ): string {
+		$permalink = get_permalink( wc_get_page_id( $page_id ) );
+
+		return $permalink ? $permalink : 'Not set';
+	}
 }

```
</details>

Reply:
Just applied this in https://github.com/Automattic/woocommerce-payments/pull/8935/commits/4e158e4e8d895cfca5c93c99cc86ded47b11c8b3. It's a great suggestion, thank you.